### ISSUE TITLE
Crash in mojo::GaneshSurface::GaneshSurface

### ISSUE THREAD

**Issue by tonygentilcore** [flutter/engine#1326]
I'm trying to directly draw to a canvas in Mojo Shell via the code here:
https://codereview.chromium.org/1368543002/

To repro this in a mojo checkout:
```
git cl patch 1368543002
cd examples/dart/camera_roll
pub get
pub run sky_tools build
pub run sky_tools run_mojo --mojo-path=../../.. --android
```

The portion of the stack that I have (with @jamesr 's help) is:
```
$ ./mojo/devtools/common/mojo_debug device stack
Inferring the build directory path as /work/mojo/src/out/android_Debug
Inferring the Android NDK path as /work/mojo/src/third_party/android_tools/ndk
Reading native crash info from stdin
Reading symbols from ['/work/mojo/src/out/android_Debug']
Using toolchain from :/work/mojo/src/third_party/android_tools/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-
signal 11 (SIGSEGV), code 1, fault addr 0x0 in tid 3687 (app_thread_0135)
pid: 3610, tid: 3687, name: app_thread_0135  >>> org.chromium.mojo.shell <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
     r0 00000000  r1 00000000  r2 a2c9fb4d  r3 00000000
     r4 00000000  r5 00000000  r6 a0b9a390  r7 a0b9a314
     r8 b4b32168  r9 a0b9a420  sl a0b9a560  fp a1e16400
     ip b6e8a6dc  sp a0b9a310  lr a2afaf8b  pc a029c8f0

Stack Trace:
  RELADDR   FUNCTION                                                                                                                       FILE:LINE
  002768f0  mojo::GaneshSurface::GaneshSurface(mojo::GaneshContext*, scoped_ptr<mojo::GLTexture, base::DefaultDeleter<mojo::GLTexture> >)  /src/flutter/engine/src/out/android_Release/../../mojo/skia/ganesh_surface.cc:25
  000a8f87  gpu::gles2::GLES2Implementation::SingleThreadChecker::~SingleThreadChecker()                                                   /work/mojo/src/out/android_Debug/../../gpu/command_buffer/client/gles2_implementation.cc:64
```

**Comment by j9brown** [comment]
I've been seeing a similar problem when running the Sky Stocks app in Mojo
shell but Mine Digger works.  Everything was fine last week.

**Comment by jamesr** [comment]
GaneshSurface talks to OpenGL from its constructor.  If this object is being created on the thread running Dart that's likely the issue as this context is only meant to be usable from Sky's GL thread.

**Comment by jamesr** [comment]
This might be a dupe of #1337

**Comment by jamesr** [comment]
With a local fix for #1337 I'm no longer seeing this crash (although nothing shows up on my device either, not sure if that's a bug with my fix or with the sample or something else).

**Comment by jamesr** [comment]
I pushed a fix for #1337 and with a local sky build including that fix I no longer get a crash in camera roll but the screen is just black.  I'm not sure if that's me holding the app wrong or another bug.

**Comment by jamesr** [comment]
I believe the fix for #1337 fixed this.  Please reopen if that's not the case.
